test-framework/kubernetes-client/src/main/java/io/quarkus/test/kubernetes/client/AbstractNamespaceCreatingTestResource.java

```diff
@@ -0,0 +1,99 @@
+package io.quarkus.test.kubernetes.client;
+
+import java.lang.annotation.Annotation;
+import java.net.ConnectException;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.awaitility.Awaitility;
+
+import io.fabric8.kubernetes.api.model.NamespaceBuilder;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+
+public abstract class AbstractNamespaceCreatingTestResource extends AbstractNamespaceConnectingTestResource {
+
+    private boolean createdNamespace = false;
+    private Context context;
+    //    private boolean preserveNamespaceOnError; todo: ns preservation
+
+    @Override
+    public void setContext(Context context) {
+        this.context = context;
+    }
+
+    @Override
+    public Map<String, String> start() {
+        createNamespaceIfRequested();
+        createdNamespace = true;
+
+        return doStart();
+    }
+
+    @Override
+    public void stop() {
+        final var client = client();
+
+        if (createdNamespace) {
+            // todo: add namespace preservation on error if/when quarkusio/quarkus#25905 becomes available
+            final var namespace = namespace();
+            logger().info("Deleting namespace '{}'", namespace);
+            client.namespaces().withName(namespace).delete();
```

Comment:
In environments like Openshift creating / deleting namespaces is not allowed for all users. Maybe some error handling is needed.